src/mlpack/methods/ann/layer/mean_pooling_impl.hpp

```diff
@@ -83,8 +123,31 @@ void MeanPooling<InputDataType, OutputDataType>::Forward(
   outputTemp = arma::zeros<arma::Cube<eT> >(outputWidth, outputHeight,
       batchSize * inSize);
 
-  for (size_t s = 0; s < inputTemp.n_slices; s++)
+  bool isPadded {padWLeft != 0 || padWRight != 0 ||
+      padHTop != 0 || padHBottom != 0};
+
+  if (isPadded)
+  {
+    inputPaddedTemp.set_size(inputTemp.n_rows + padWLeft + padWRight,
+        inputTemp.n_cols + padHTop + padHBottom, inputTemp.n_slices);
+
+    for (size_t i = 0; i < inputTemp.n_slices; ++i)
+    {
+      padding.Forward(std::move(inputTemp.slice(i)),
+          std::move(inputPaddedTemp.slice(i)));
+    }
+  }
+
+  if (isPadded)
+  {
+    for (size_t s = 0; s < inputPaddedTemp.n_slices; s++)
+    Pooling(inputPaddedTemp.slice(s), outputTemp.slice(s));
```

Comment:
[Style Issue] Also could you place the condition inside braces.

Reply:
Not sure which condition you mean. You mean like `(s < inputPaddedTemp.n_slices)` ? I haven't seen this style in rest of the code.

Reply:
Like 
```
for (size_t s = 0; s < inputPaddedTemp.n_slices; s++)
 {
    Pooling(inputPaddedTemp.slice(s), outputTemp.slice(s));
 }
```

Reply:
Oh, got that.